### PR TITLE
APIMF-2963: define allowedPlugins and use for filtering in parse plugin selection

### PR DESCRIPTION
**Pull request by AgustinBettati**

### CODE REVIEW

File: shared/src/main/scala/amf/client/remod/amfcore/plugins/parse/AMFParsePlugin.scala [hunk 1]
```diff
@@ -12,6 +12,16 @@ private[amf] trait AMFParsePlugin extends AMFPlugin[ParsingInfo] {
 //  def parse(document:Root, ctx:ParserContext): BaseUnit // change parser for AMF context
   def parse(document: Root, ctx: ParserContext, options: ParsingOptions): BaseUnit
 
+  /**
+    * vendors that are parsed by this plugin.
+    */
+  def vendors: Seq[String]
```
Comment (nschejtman) [inline shared/src/main/scala/amf/client/remod/amfcore/plugins/parse/AMFParsePlugin.scala]: mediaTypes?

File: shared/src/main/scala/amf/core/parser/Reference.scala [hunk 2]
```diff
@@ -22,31 +23,35 @@ case class Reference(url: String, refs: Seq[RefContainer]) extends PlatformSecre
     copy(refs = refs :+ RefContainer(kind, ast, fragment))
   }
 
-  def resolve(compilerContext: CompilerContext, allowRecursiveRefs: Boolean)(
+  def resolve(compilerContext: CompilerContext, domainPlugin: AMFParsePlugin, allowRecursiveRefs: Boolean)(
```
Comment (nschejtman) [inline shared/src/main/scala/amf/core/parser/Reference.scala]: Passing the list of allowed media types as parameter seems clearer to me. Passing a parse plugin makes me assume that plugin will parse the reference.

File: shared/src/main/scala/amf/client/remod/amfcore/plugins/parse/AMFParsePlugin.scala (hunk 1 above)
Comment (AgustinBettati) [reply shared/src/main/scala/amf/client/remod/amfcore/plugins/parse/AMFParsePlugin.scala]: adjusted to new convention.

File: shared/src/main/scala/amf/core/parser/Reference.scala (hunk 2 above)
Comment (AgustinBettati) [reply shared/src/main/scala/amf/core/parser/Reference.scala]: Agreed.

**Review by nschejtman — DISMISSED** [review]

**Review by nschejtman — APPROVED** [review]

**Pull request merged**